app/src/main/java/org/mozilla/fenix/ext/BookmarkNode.kt

```diff
@@ -15,5 +15,9 @@ val Context.bookmarkStorage: PlacesBookmarksStorage
  * Removes [children] from [BookmarkNode.children] and returns the new modified [BookmarkNode].
  */
 operator fun BookmarkNode.minus(children: Set<BookmarkNode>): BookmarkNode {
-    return this.copy(children = this.children?.filter { it !in children })
+    return this.copy(children = this.children?.filter { filtered ->
+        children.none {
```

Comment:
Could you convert `children` into a set of guids to avoid the nested loop? I think it would be easier to read that way too.

Reply:
sure!